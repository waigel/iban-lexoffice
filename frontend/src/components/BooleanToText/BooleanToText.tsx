import {useTranslate} from '@tolgee/react';
import React from 'react';

export interface BooleanToTextProps {
  value: boolean;
}
export const BooleanToText = ({value}: BooleanToTextProps) => {
  const {t} = useTranslate();
  return (
    <span className={value ? 'text-success' : 'text-red-500'}>
      {value ? t('YES') : t('NO')}
    </span>
  );
};
