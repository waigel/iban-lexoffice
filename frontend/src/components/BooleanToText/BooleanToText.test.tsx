import React from 'react';
import TestRenderer from 'react-test-renderer';
import {BooleanToText} from './BooleanToText';

jest.mock('@tolgee/react', () => ({
  useTranslate: () => ({t: jest.fn(key => key), isLoading: false}),
}));

it('should test the BooleanToText component is rendered correctly for true', () => {
  const tree = TestRenderer.create(<BooleanToText value />);
  expect(tree.toJSON()).toMatchSnapshot();
});

it('should test the BooleanToText component is rendered correctly for false', () => {
  const tree = TestRenderer.create(<BooleanToText value={false} />);
  expect(tree.toJSON()).toMatchSnapshot();
});
