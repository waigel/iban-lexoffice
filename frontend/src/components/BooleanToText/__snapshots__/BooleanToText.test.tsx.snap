// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the BooleanToText component is rendered correctly for false 1`] = `
<span
  className="text-red-500"
>
  NO
</span>
`;

exports[`should test the BooleanToText component is rendered correctly for true 1`] = `
<span
  className="text-success"
>
  YES
</span>
`;
